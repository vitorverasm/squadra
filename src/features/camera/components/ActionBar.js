import React from 'react';
import { StyleSheet, TouchableHighlight, View } from 'react-native';
import { colors } from '../../../utils/styles';

const ActionBar = ({ takePhoto }) => (
  <View
    style={{
      flex: 0,
      position: 'absolute',
      bottom: 0,
      height: 100,
      width: '100%',
      flexDirection: 'row',
      justifyContent: 'center',
      backgroundColor: 'transparent'
    }}
  >
    <View style={{ flex: 1 }} />
    <TouchableHighlight
      onPress={takePhoto}
      style={styles.captureButton}
      underlayColor={colors.crimsonRed}
      activeOpacity={0}
    >
      <View style={{ flex: 1, borderRadius: 70, backgroundColor: colors.primary }} />
    </TouchableHighlight>
    <View style={{ flex: 1 }} />
  </View>
);

const styles = StyleSheet.create({
  captureButton: {
    flex: 0,
    backgroundColor: 'transparent',
    borderColor: colors.white,
    borderWidth: 2,
    height: 70,
    width: 70,
    borderRadius: 70,
    alignSelf: 'center'
  }
});

export default ActionBar;
