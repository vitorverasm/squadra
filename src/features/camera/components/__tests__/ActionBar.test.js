import { shallow } from 'enzyme';
import React from 'react';
import ActionBar from '../ActionBar';

describe('ActionBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ActionBar takePhoto={jest.fn} toggleFlash={jest.fn} goToAlbums={jest.fn} flashOn />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
