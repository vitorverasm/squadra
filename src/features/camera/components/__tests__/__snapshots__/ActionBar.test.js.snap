// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar renders correctly 1`] = `
"<View style={{...}}>
  <View style={{...}}>
    <Connect(TagSelector) />
  </View>
  <View style={{...}}>
    <TouchableOpacity style={{...}} onPress={[Function: bound fn]} activeOpacity={0.2}>
      <Icon name=\\"zap\\" size={30} color=\\"#FF3B30\\" style={{...}} allowFontScaling={false} />
      <Text style={{...}}>
        ON
      </Text>
    </TouchableOpacity>
    <TouchableHighlight onPress={[Function: bound fn]} style={{...}} underlayColor=\\"#A50B25\\" activeOpacity={0} delayPressOut={100}>
      <View style={{...}} />
    </TouchableHighlight>
    <TouchableOpacity style={{...}} onPress={[Function: bound fn]} activeOpacity={0.2}>
      <Icon name=\\"image\\" size={30} color=\\"#FFFFFF\\" style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
</View>"
`;
